docs/expression_syntax.md

```diff
@@ -0,0 +1,195 @@
+**Is your feature request related to a problem? Please describe.**
+As part of a larger feature (#522) to support complex condition statements in data prepper there is a need to define a syntax for conditional statements. A conditional statement is a String that is evaluated at runtime and may reference fields within a record.
+
+**Describe the solution you'd like**
+_Terms used throught this document are defined in the Definitions section_
+
+## Supported Operators
+In order of evaluation priority. _(top to bottom, left to right)_
+
+| Operator             | Description              | Data Prepper Version |
```

Comment:
Can we have this priority table in an increasing/decreasing order of Data prepper version?

Reply:
I think the real problem is the table has too much information. The primary goal of this section is to document order of operations. Removing the version and including only supported operators for 1.3.0 will help here.

Similar to the Java Operator Precedence table
<img width="582" alt="image" src="https://user-images.githubusercontent.com/8837988/157697680-13230e29-1623-4562-8315-bc35ac81e9fb.png">

@sshivanii let me know your thoughts on this.
@dlvenable you also commented, I'd like to hear your input also.

Reply:
@sbayer55 I like this Java Operator Precedence approach, we can have the supported operators for only v1.3.0 and update the table based on more operators as and when we add it.

Reply:
I updated the table to fit this suggestion.